client/a8c-for-agencies/components/layout/banner.tsx

```diff
@@ -9,14 +12,42 @@ type Props = {
 	level: 'error' | 'warning' | 'info' | 'success';
 	onClose?: () => void;
 	title?: string;
+	preferenceName?: string;
 };
 
-export default function LayoutBanner( { className, children, onClose, title, actions }: Props ) {
+export default function LayoutBanner( {
+	className,
+	children,
+	onClose,
+	title,
+	actions,
+	level = 'success',
+	preferenceName,
+}: Props ) {
+	const dispatch = useDispatch();
+
+	const bannerShown = useSelector( ( state ) =>
+		preferenceName ? getPreference( state, preferenceName ) : false
+	);
+
 	const wrapperClass = clsx( className, 'a4a-layout__banner' );
 
+	const handleClose = () => {
+		if ( preferenceName ) {
+			dispatch( savePreference( preferenceName, true ) );
+		}
+		if ( onClose ) {
+			onClose();
+		}
```

Comment:
Nitpicking: We can do it like below with a lesser line.

```suggestion
		onClose?.();
```

Reply:
Ah, yeah. Fixed it: https://github.com/Automattic/wp-calypso/compare/da30da52ecb8af8c6ccbda90ffd4a527f2f0701e..d8a7797aca9a112f6e07b4489cbc706d077969f7